pymeasure/instruments/rohdeschwarz/hmp.py

```diff
@@ -0,0 +1,288 @@
+        values=[1, 2, 3, 4],
+        get_process=lambda x: int(x),
+    )
+
+    # Implemented as a more natural way of selecting a channel.
+    def select_channel(self, channel):
+        """Select a channel."""
+        self.selected_channel = channel
+
+    # Voltage Settings --------------------------------------------------------
+
+    voltage = Instrument.control(
+        "VOLT?", "VOLT %s", "Output voltage in V. Increment 0.001 V."
+    )
+
+    minimum_voltage = Instrument.measurement(
+        "VOLT? MIN", "Minimum voltage in V."
+    )
+
+    max_voltage = Instrument.measurement(
+        "VOLT? MAX", "Maximum voltage in V."
+    )
+
+    def voltage_to_min(self):
+        """Set voltage of the selected channel to its minimum value."""
+        self.write("VOLT MIN")
+
+    def voltage_to_max(self):
+        """Set voltage of the selected channel to its maximum value."""
+        self.write("VOLT MAX")
+
+    voltage_step = Instrument.control(
+        "VOLT:STEP?",
+        "VOLT:STEP %s",
+        "Voltage step in V. Default 1 V.",
+        validator=truncated_range,
+        values=[0, 32.050],
+    )
+
+    def voltage_up(self):
+        """Increase voltage by one step."""
+        self.write("VOLT UP")
+
+    def voltage_down(self):
+        """Decrease voltage by one step."""
+        self.write("VOLT DOWN")
+
+    # Current Settings --------------------------------------------------------
+
+    current = Instrument.control(
+        "CURR?",
+        "CURR %s",
+        "Output current in A. Range depends on instrument type.",
+    )
+
+    min_current = Instrument.measurement(
+        "CURR? MIN", "Minimum current in A."
+    )
+
+    max_current = Instrument.measurement(
+        "CURR? MAX", "Maximum current in A."
+    )
+
+    def current_to_min(self):
+        """Set current of the selected channel to its minimum value."""
+        self.write("CURR MIN")
+
+    def current_to_max(self):
+        """Set current of the selected channel to its maximum value."""
+        self.write("CURR MAX")
+
+    current_step = Instrument.control(
+        "CURR:STEP?", "CURR:STEP %s", "Current step in A."
+    )
+
+    def current_up(self):
+        """Increase current by one step."""
+        self.write("CURR UP")
+
+    def current_down(self):
+        """Decreases current by one step."""
+        self.write("CURR DOWN")
+
+    # Combined Voltage And Current Settings -----------------------------------
+
+    voltage_and_current = Instrument.control(
+        "APPL?",
+        "APPL %s, %s",
+        "Output voltage (V) and current (A).",
+    )
+
+    # Output Settings ---------------------------------------------------------
+
+    selected_channel_active = Instrument.control(
+        "OUTP:SEL?",
+        "OUTPUT:SEL %s",
+        "Set the selected channel to active or inactive or check its status.",
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    output_on = Instrument.control(
```

Comment:
`output_enabled` would echo some other instruments that use that name already (there is none that has `output_on`)